modules/maven/src/test/java/org/eclipse/sw360/antenna/maven/workflow/processors/MavenArtifactResolverTest.java

```diff
@@ -0,0 +1,136 @@
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.mockito.internal.verification.VerificationModeFactory.atLeast;
+
+public class MavenArtifactResolverTest extends AntennaTestWithMockedContext {
+    private IProject project;
+    private MavenArtifactResolver mavenArtifactResolver;
+    private List<Artifact> artifacts;
+
+    @Before
+    public void setUp() throws IOException {
+        System.setProperty("maven.home", "/usr/share/maven");
+        project = Mockito.mock(IProject.class);
+        when(antennaContextMock.getProject()).thenReturn(project);
+        org.apache.maven.repository.RepositorySystem mvnRepositorySystem = null;
+        when(antennaContextMock.getGeneric(org.apache.maven.repository.RepositorySystem.class))
+                .thenReturn(Optional.ofNullable(mvnRepositorySystem));
+        MavenProject mavenProject = new MavenProject();
+        when(antennaContextMock.getGeneric(MavenProject.class))
+                .thenReturn(Optional.of(mavenProject));
+        LegacySupport legacySupport = null;
+        when(antennaContextMock.getGeneric(LegacySupport.class))
+                .thenReturn(Optional.ofNullable(legacySupport));
+
+        when(configMock.getIgnoreForSourceResolving()).thenReturn(Collections.emptyList());
+        when(project.getBasedir()).thenReturn(new File(""));
+        when(toolConfigMock.getAntennaTargetDirectory()).thenReturn(temporaryFolder.newFolder("target").toPath());
+        Path targetDir = toolConfigMock.getAntennaTargetDirectory();
+        when(toolConfigMock.getDependenciesDirectory()).thenReturn(targetDir.resolve("dependencies"));
+
+        mavenArtifactResolver = new MavenArtifactResolver();
+        mavenArtifactResolver.setAntennaContext(antennaContextMock);
+
+        artifacts = Collections.singletonList(mkArtifact("commons-csv", "org.apache.commons", "1.7"));
+    }
+
+    @Test
+    public void mavenArtifactResolverForGoodArtifact() throws AntennaException {
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("sourceRepositoryUrl", "");
+        configMap.put("preferredSourceQualifier", "sources-ext");
+        mavenArtifactResolver.configure(configMap);
+
+        Collection<Artifact> processedArtifacts = mavenArtifactResolver.process(artifacts);
+
+        assertThat(processedArtifacts.size()).isEqualTo(1);
+        assertThat(antennaContextMock.getToolConfiguration().getDependenciesDirectory().resolve("commons-csv-1.7.jar")).exists();
+    }
+
+    @Test
+    public void mavenArtifactResolverForGoodArtifactDirectWithImpl() throws MalformedURLException, AntennaException {
+        ProxySettings proxySettings = new ProxySettings(false, null, 0);
+
+        ContextExtension contextExtension = new ContextExtension();
+
+        Collection<Artifact> processedArtifacts = new MavenArtifactResolverImpl(proxySettings, contextExtension.get(RepositorySystem.class),
```

Comment:
usually a test consists of three blocks: initialization, execution and assertion. This section here mixes both, initialization and execution. Maybe create a variable for the `MavenArtifactResolverImpl` first.